### PR TITLE
implement set_cursor_hittest for x11

### PR DESCRIPTION
**Pull request by lunixbochs**
- [x] Tested on all platforms changed
- [x] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [x] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [x] Created or updated an example program if it would help users understand this functionality
- [x] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

### CODE REVIEW

File: src/platform_impl/linux/x11/window.rs [hunk 1]
```diff
@@ -1449,8 +1449,37 @@ impl UnownedWindow {
...
+                width: size.width as u16,
+                height: size.height as u16,
+            })
+        }
+        unsafe {
+            let region = (self.xconn.xfixes.XFixesCreateRegion)(
+                self.xconn.display,
+                rectangles.as_mut_ptr(),
+                rectangles.len() as i32,
+            );
```
Comment (notgull) [inline src/platform_impl/linux/x11/window.rs:1469]: Can this region be cached?

Comment (lunixbochs) [reply src/platform_impl/linux/x11/window.rs:1469]: Where? It has the window size in it. Also at least in my use case, I'm usually just disabling hittest once and leaving it disabled for any given window.

**Comment by lunixbochs** [comment]
this has an issue - if the window is resized, the hittest region becomes invalid, so we need to reapply it

**Comment by kchibisov** [comment]
ping @notgull

**Review by notgull — CHANGES_REQUESTED** [review]
Looks good to me, but this needs to be rebased on `master`. Probably using the [`x11rb` equivalent](https://docs.rs/x11rb/latest/x11rb/protocol/xfixes/fn.set_window_shape_region.html).

**Comment by lunixbochs** [comment]
I rebased and switched to xcb. Could use testing and my app doesn't target winit master atm

**Review by notgull — APPROVED** [review]
Thanks!

**Pull request merged**